draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -136,6 +136,65 @@ This document uses "service" and "workload" interchangeably. Otherwise, all term
 
 ## Option 2: Authentication Based on HTTP Message Signatures
 
+This option uses the WIMSE Identity Token (ref TBD) to sign the request and optionally, the response.
+This specification defines a profile of the Message Signatures specification {{!RFC9421}}.
+
+The request is signed as per {{RFC9421}}. The following derived components MUST be signed:
+
+* `@method`
+* `@request-target`
+
+In addition, the following headers MUST be signed when they exist:
+
+* `content-type`
+* `content-digest`
+* `Authorization`
+* TBD that includes the WIT
+
+If the response is signed, the following components MUST be signed:
+
+* `@status`
+* `@method;req`
+* `@request-target;req`
+* `content-type` if it exists
+* `content-digest` if it exists
+* `Authorization`
+* TBD that includes the WIT
+
+For both requests and responses, the following signature parameters MUST be included:
+
+* `created`
+* `expires` - expiration MUST be short, e.g. on the order of minutes. The WIMSE architecture will provide seperate
+mechanisms in support of long-lived compute processes.
+* `nonce`
+* `alg`
+* `tag` - the value for implementations of this specification is `wimse-service-to-service`
+
+Since the signing key is sent along with the message, the `keyid` parameter SHOULD NOT be used.
+
+It is RECOMMENDED to include only one signature with the HTTP message.
+If multiple ones are included, then the signature label included in both the `Signature-Input` and `Signature` headers SHOULD
+be `wimse`.
+
+To promote interoperability, the `ecdsa-p256-sha256` signing algorithm MUST be implemented
+by general purpose implementations of this spec.
+
+The `Accept-Signature` field is not used by this profile
+
+Following is a non-normative example of a signed request and a signed response, using the keys mentioned in Section TBD.
+
+~~~ http
+GET /gimme-ice-cream?flavor=vanilla HTTP/1.1
+Host: example.com
+Authorization: Basic c3BpZmZlOi8vcmVhbG0uZXhhbXBsZS5jb20vc3ZjYmMxMjM=
+Signature: wimse=:cJktyi42FuhfTjuJl5vlwHMkQggviWQD/ho1UKMy96VtOeECRrc \
+UhxlabRRaTgXdRJrzVR7IAfvR7/8E4y7yBg==:
+Signature-Input: wimse=("@method" "@request-target" "authorization"); \
+created=1717799748;expires=1717800048; \
+nonce="abcd1111";tag="wimse-service-to-service";keyid="dummy-keyid-TODO"
```

Comment:
above it says that, "the `keyid` parameter SHOULD NOT be used." but there's a keyid in this line of the example

Reply:
also says alg is required above but doesn't seem to be in the example

Reply:
> above it says that, "the `keyid` parameter SHOULD NOT be used." but there's a keyid in this line of the example

This was a problem with my own implementation of Message Sigs, now fixed.

Reply:
> also says `alg` is required above but doesn't seem to be in the example

In fact I think that `alg` as a Message Sig parameter is a mistake and a vulnerability waiting to happen (because verifiers would trust this field even though they lack context, similarly to the "none" issue with JWTs. IMO an algorithm should be strongly bound to a key. Can we make the `alg` field mandatory within the JWK?

Reply:
`keyid` comment addressed. `alg` comment now #16.